CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 cmake_minimum_required( VERSION 3.16 )
 
+if(POLICY CMP0144)
+  cmake_policy(SET CMP0144 NEW)   # avoid cmake warning, accept both lowercase and uppercase `<PackageName>_ROOT`
+endif()
```

Comment:
What is responsible for that warning? Something we're doing or something in boost? I wonder if doing `cmake_minimum_required( VERSION 3.16...4.0 )` is better (though that opts us in to a _lot_ more new policies, perhaps we'd not want to opt in to any policies for a version we don't test in CI, but that's no real problem since we can move the cmake in reproducible build to anywhere we'd like)

Reply:
No, noth8ing to do with boost, it is because we have `find_package( cdt REQUIRED )` in a couple of `CMakeLists.txt`, and because `cdt` is in lower case. I get the warning with `cmake 3.29.6`

Reply:
> I wonder if doing `cmake_minimum_required( VERSION 3.16...4.0 )` is better

I think we'd have to go higher than that because `CMP0144` was introduced in cmake `3.27`.

https://cmake.org/cmake/help/latest/policy/CMP0144.html

Reply:
Also, can we get rid of that `EOSIO_CDT` stuff?
https://github.com/AntelopeIO/spring/blob/e08e1f2a197acc88987ca17332c44916735cd883/unittests/CMakeLists.txt#L5-L18

Reply:
> > I wonder if doing `cmake_minimum_required( VERSION 3.16...4.0 )` is better
> 
> I think we'd have to go higher than that because `CMP0144` was introduced in cmake `3.27`.
> 
> https://cmake.org/cmake/help/latest/policy/CMP0144.html

When you do `<min>...<max>` it enables all policies up to `max`. So on 3.27+ it would enable CMP0144 by default (and a bunch of others).

Reply:
> When you do `<min>...<max>` it enables all policies up to `max`. So on 3.27+ it would enable CMP0144 by default (and a bunch of others).

Sounds good to me. I'll do that!